src/expr/src/relation/mod.rs

```diff
@@ -2769,7 +2769,23 @@ impl RowSetFinishing {
             })
             .unwrap_or(limit);
 
-        let mut ret = Vec::with_capacity(return_size);
+        // Bail early if we know our results won't fit into our Vec
+        // 
+        // Note: this check is a _lower bound_ as to how much memory
+        // we'll need for results. Calculating the actual answer would
+        // require us to iterate through all of the rows
+        if return_size * std::mem::size_of::<Row>() > max_result_size {
```

Comment:
I think this multiplication can overflow, no?

Reply:
Great call, updated this to use `saturating_mul` which will saturate at the `usize::MAX`